### DISCUSSION TITLE
FastCGI support

### DISCUSSION THREAD

**Discussion by danrabbit** [curl/curl#14993] [Ideas]
Lately I've seen more projects use FastCGI protocol (aside from PHP using it for a while now). All major web servers/load balancers support it. It is a super fast binary protocol that can be used for non-secure communications behind a reverse proxy or on private networks. It would be useful for Curl to support FastCGI protocol on the client side.

**Comment by dfandrich** [comment]
FastCGI is a server-side protocol, but curl is entirely client-side. How would you envision such protocol support working?

**Comment by foudfou** [comment]
I understand the client-side argument. Just to explain where the request comes from (speaking under the control of @danrabbit): lately I was wanting to test a fresh php-fpm install, from the point of view of the server, without installing a webserver and configuring it. And so naively I thought curl would implement some basic fastcgi functionality, maybe something like `curl --fastcgi <host> <port> <document_root> <script_filename>` (not sure if/how a URL form would make sense here).
But ok that's just me. And I understand there's probably a ton of more important things to work on right now :smile: 
:heart:
